up-l1/README.adoc

```diff
@@ -261,6 +261,38 @@ end
 deactivate T
 ----
 
+In RegisterListener API, we have the source and sink `UUri` in the arguments.
+However, not all the combinations of source and sink `UUri` are valid.
+
+The following table are the valid combinations, which maps to the corresponding user case.
+
+[%autowidth]
+|===
+| source `resource_id` | sink `resource_id` | Publish | Notification | Request | Response | Use Case
+
+| [8000-FFFE] |   None   |    V    |              |         |         | A uE listens for a specific published message 
+| [8000-FFFE] |     0    |         |      V       |         |         | A uE listens for a specific notification message
+|      0      | [1-7FFF] |         |              |    V    |         | A uE listens for a specific request 
+|   [1-7FFF]  |     0    |         |              |         |    V    | A uE listens for a specific response
+|     FFFF    |     0    |         |      V       |         |    V    | A uE listens for all notifications and responses 
+|     FFFF    |   FFFF   |         |      V       |    V    |    V    | uStreamer listens for all notifications, requests, and responses
+|===
+
+Note that other combinations not in the table *MUST* never be used in RegisterListener API.
```

Comment:
Sorry @evshary but I'm reminded of what I actually implemented in up-java. The [InMemoryRpcClient.java](https://github.com/eclipse-uprotocol/up-java/blob/main/src/main/java/org/eclipse/uprotocol/communication/InMemoryRpcClient.java) will register a listener for source=FFFF, sink=0 so that it can get all responses. Additionally, [InMemoryRpcServer.java](https://github.com/eclipse-uprotocol/up-java/blob/main/src/main/java/org/eclipse/uprotocol/communication/InMemoryRpcServer.java#L100) will register a listener with source=FFFF and sink=[1-7FFFF] to be able to receive requests from any client for a specific method so we need to add those options to the table...

Reply:
For the case `source=FFFF, sink=0`, it's covered at index 5.
However, I'm not sure about `source=FFFF, sink=[1-7FFF]`. Could we use `source=0, sink=[1-7FFF]` to cover it?
I mean we can set Authority to any, but specify the `resource_id` must be 0. It doesn't make sense for any other `resource_id`.

```
UUri {
  authority_name: "*",         // any authority
  ue_id: 0x0000_FFFF           // any instance, any service
  ue_version_major: 0xFF,  // any
  resource_id: 0
}
```

I'm fine to add it back, but in that case, should we also add the index 7-9 back? I removed them because they can be covered by other cases.
![image](https://github.com/eclipse-uprotocol/up-spec/assets/456210/04be5a7e-891f-4ac4-8ddf-d71e78fc7a35)

Reply:
Hi @stevenhartley, I forgot to tag you in the last comment. Let me know if you have any other ideas. Thank you!

Reply:
I think it is Ok for now. We will need to cleanup this API to make the API more explicit as the use of the API is still confusing (something for 1.7 or 2.0 later).